gcore/gdalalgorithm.cpp

```diff
@@ -1903,6 +1903,13 @@ bool GDALAlgorithm::ValidateArguments()
             }
         }
     }
+
+    for (const auto &f : m_validationActions)
+    {
+        if (!f())
+            ret = false;
```

Comment:
question: should we break here?

Reply:
no, the method tries to run as much validation tests as possible so that the user gets all errors at once. I'll add a comment at its beginning